trade-finance-manager-ui/server/helpers/express-session.ts

```diff
@@ -0,0 +1,31 @@
+import { Request } from 'express';
+import { UserSessionData } from '../types/express-session';
+
+type Session = Request['session'];
+
+type UserSession = Session & UserSessionData;
+
+/**
+ * By default, all session data will be optional
+ * (see use of `Partial` {@link https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/express-session/index.d.ts#L17 here})
+ *
+ * This helper function asserts that the `user` and `userToken` properties are
+ * present in the session, or throws an `Error` if not.
+ *
+ * Note: this doesn't assert on the validity of the user data values (which
+ * should have already been authenticated in middleware by this point), only
+ * that the values are present.
+ */
+export const asUserSession = (session: Session): UserSession => {
```

Comment:
Is there a reason to use `Object.assign` in the return here rather than the type predicate `Session is UserSession` and return `true` if all the checks pass? It just feels like, when more properties are added to `Session`, we need to both add the check and add the field to `Object.assign`, whereas having a `return true` at the bottom means you only need to add the check

Reply:
We could change as you suggest, and then change the function name to `isUserSession`, but then instead of just
```
const { user, userToken } = asUserSession(req.session);
// do things with user and userToken
```
we'd then have to change things to something like
```
if (isUserSession(req.session)) {
  const { user, userToken } = req.session;
  // do things with user and userToken
} else {
  throw new Error('expected user session...');
}
```
which is a lot more verbose. The likelihood of needing to update this function to add more properties seems quite small, but if we do that seems like less work that having to add the extra code every where this is used.

Let me know though if you had something else in mind.

Reply:
That makes sense to me